src/code/z_play.c

```diff
@@ -1,59 +1,1464 @@
+               (gSaveContext.sceneSetupIndex < 4)) {
+        if (gSaveContext.eventChkInf[4] & 0x100) {
+            gSaveContext.sceneSetupIndex = 3;
+        } else {
+            gSaveContext.sceneSetupIndex = 2;
+        }
+    }
+
+    spawnEntrance = &gEntranceTable[gSaveContext.entranceIndex + gSaveContext.sceneSetupIndex];
+    Gameplay_SpawnScene(globalCtx, spawnEntrance->scene, spawnEntrance->spawn);
+    osSyncPrintf("\nSCENE_NO=%d COUNTER=%d\n", gSaveContext.entranceIndex, gSaveContext.sceneSetupIndex);
+
+    // When entering Gerudo Valley in the right setup, trigger the GC emulator to play the ending movie.
+    // The emulator constantly checks whether PC is 0x81000000, so this works even though it's not a valid address.
+    if ((gEntranceTable[gSaveContext.entranceIndex].scene == SCENE_SPOT09) && (gSaveContext.sceneSetupIndex == 6)) {
+        osSyncPrintf("エンディングはじまるよー\n"); // "The ending starts"
+        ((void (*)())0x81000000)();
+        osSyncPrintf("出戻り？\n"); // "Return?"
+    }
+
+    Cutscene_HandleEntranceTriggers(globalCtx);
+    KaleidoScopeCall_Init(globalCtx);
+    func_801109B0(globalCtx);
+
+    if (gSaveContext.nextDayTime != 0xFFFF) {
+        if (gSaveContext.nextDayTime == 0x8001) {
+            gSaveContext.unk_14++;
+            gSaveContext.unk_18++;
+            gSaveContext.dogIsLost = true;
+            if (Inventory_ReplaceItem(globalCtx, ITEM_WEIRD_EGG, ITEM_CHICKEN) ||
+                Inventory_ReplaceItem(globalCtx, ITEM_POCKET_EGG, ITEM_POCKET_CUCCO)) {
+                func_8010B680(globalCtx, 0x3066, 0);
+            }
+            gSaveContext.nextDayTime = 0xFFFE;
+        } else {
+            gSaveContext.nextDayTime = 0xFFFD;
+        }
+    }
+
+    SREG(91) = -1;
+    R_PAUSE_MENU_MODE = 0;
+    func_800C0EA8(&globalCtx->preRenderCtx);
+    func_800C0E70(&globalCtx->preRenderCtx, 0x140, 0xF0, 0, 0, 0);
+    func_800C0ED8(&globalCtx->preRenderCtx, 0x140, 0xF0, 0, 0);
+    D_80161490 = 0;
+    globalCtx->unk_1241B = 0;
+    func_8008E6A0(&globalCtx->sub_7B8);
+    func_800FD9A0((u32)osGetTime());
+    Matrix_Init(&globalCtx->state);
+    globalCtx->state.main = Gameplay_Main;
+    globalCtx->state.destroy = Gameplay_Destroy;
+    globalCtx->sceneLoadFlag = -0x14;
+    globalCtx->unk_11E16 = 0xFF;
+    globalCtx->unk_11E18 = 0;
+    globalCtx->unk_11DE9 = 0;
+
+    if (gSaveContext.gameMode != 1) {
+        if (gSaveContext.nextTransition == 0xFF) {
+            globalCtx->fadeTransition =
+                (gEntranceTable[gSaveContext.entranceIndex + tempSetupIndex].field >> 7) & 0x7F; // Fade In
+        } else {
+            globalCtx->fadeTransition = gSaveContext.nextTransition;
+            gSaveContext.nextTransition = 0xFF;
+        }
+    } else {
+        globalCtx->fadeTransition = 6;
+    }
+
+    func_800B3908();
+    func_800B3538(&globalCtx->sub_1241C);
+    func_800B37EC(&globalCtx->sub_1241C, 3);
+    func_800B37E4(&globalCtx->sub_1241C, 0xA0A0A0FF);
```

Comment:
Should we have a macro for packed RGBA colors? To make it clear that this is intended to be a color.
```suggestion
    #define GPACK_RGBA8888(r,g,b,a) (((r & 0xFF) << 24)|((g & 0xFF) << 16)|((b & 0xFF) << 8) |((a & 0xFF) << 0))
    func_800B37E4(&globalCtx->sub_1241C, GPACK_RGBA8888(0xA0,0xA0,0xA0,0xFF));
```

Reply:
Yeah this is the first time I encounter byte packed colors. I considered it and was going to wait a bit before adding a macro for it, but I'll just do it now.